lib/core.js

```diff
@@ -407,7 +407,7 @@ declare class Promise<R> {
 
     then<U>(
       onFulfill?: (value: R) => Promise<U> | U,
-      onReject?: (error: any) => Promise<U> | U
+      onReject?: (error: any) => ?Promise<U> | U
```

Comment:
The spec is rather hard to read, but here they state that `promise.catch(onReject)` is exactly equivalent to `promise.then(undefined, onReject)`:
http://www.ecma-international.org/ecma-262/6.0/#sec-promise.prototype.catch

Also check https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/es6-promise/es6-promise.d.ts -- they explicitly allow the "reject" part to return nothing. This is actually the most common use case:
```
promise.then((data) => {
  return transform(data);
}).catch((error) => {
  console.log("There was an error handling the promise. Abort!");
  // Returning nothing 
});
```